Make Synapse Core the CODEOWNERS of templated repositories if a `synapse_team` variant is in use. (#8)
matrix-org/synapse-module-cookiecutter-template: README.md

    Synapse 1.46 and later)
    * strict by default (configurable in `mypy.ini`)
* unit testing using `trial` (invokable with `tox`, checked in CI)

> * *Specific to the 'Synapse Team' variant:*

Reviewer: We seem to have a mix of 'Synapse Team' and 'Synapse **Core** Team'.

Author: 'Synapse Core' refers to the team name on GitHub, but yeah it's not super consistent

Reviewer: Ah, I see OK. That is a little confusing indeed, but the single quotes do help.

>   * `CODEOWNERS` for review requests on Synapse Core Team

Reviewer: ```suggestion
  * `CODEOWNERS` for automatic review requests for the Synapse Core Team when a
    pull request is opened
```

   Although this template does not initialise a Git repository automatically,
   the CI configuration and suggested release instructions mention it.
   This is commonly `main`, `master`, `trunk`, `default` or `develop`.

> 7. `variant`: Choose whether or not to include the Synapse Team-specific
>    customisations; use `standard` if you are a community member or `synapse_team`
>    if the Synapse Team is going to be maintaining this module.

Reviewer: I wonder if "community member" is the right language. I may not consider myself part of the Matrix "community" -- but I do know I'm not part of the Synapse maintenance team.

Perhaps: ?

```suggestion
7. `variant`: Choose whether or not to include the Synapse Team-specific
   customisations; pick `synapse_team` if the module will be maintained
   by the Synapse Team. Otherwise, pick `standard`.
```

Author: I'm wondering how this will fit in with customer work;
some customer work is generic enough that we may choose to release it (e.g. https://github.com/matrix-org/synapse-user-restrictions/) at which point it's maintained by the Synapse Team (I guess?).

I don't know about private modules we keep for customers in GitLab; right now they would probably make most sense to be `standard` (since they don't need a CODEOWNERS and they don't need Apache 2.0 licence), but I can see it happening that we make more Element-specific tweaks that *do* make sense for both private and public projects.

For now, though, I think you're right. If we need to make further distinctions down the line, we can cross that bridge then.

   Although this template does not initialise a Git repository automatically,
   the CI configuration and suggested release instructions mention it.
   This is commonly `main`, `master`, `trunk`, `default` or `develop`.

> 7. `variant`: Choose whether or not to include the Synapse Team-specific

Reviewer: ```suggestion
7. `variant`: Choose whether or not to include the 'Synapse Team'-specific
```
